ci-operator/step-registry/cucushift/upgrade/checktarget/cucushift-upgrade-checktarget-commands.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+
+# Extract oc binary which is supposed to be identical with target release
+function extract_oc(){
+    echo -e "Extracting oc\n"
+    local retry=5 tmp_oc="/tmp/client-2"
+    mkdir -p ${tmp_oc}
+    while ! (env "NO_PROXY=*" "no_proxy=*" oc adm release extract -a "${CLUSTER_PROFILE_DIR}/pull-secret" --command=oc --to=${tmp_oc} ${TARGET});
+    do
+        echo >&2 "Failed to extract oc binary, retry..."
+        (( retry -= 1 ))
+        if (( retry < 0 )); then return 1; fi
+        sleep 60
+    done
+    mv ${tmp_oc}/oc ${OC_DIR} -f
+    which oc
+    oc version --client
+    return 0
+}
+
+function check_target_version() {
+    local conditional_updates=$(oc get clusterversion version -o json | jq -r '.status.conditionalUpdates[]?.release.version' | xargs)
+    if [[ " $conditionalUpdates " =~ " $TARGET_VERSION " ]]; then
+        echo "Error: $TARGET_VERSION is not recommended, for details please refer:"
+        oc get clusterversion version -o json | jq -r '.status.conditionalUpdates'
+        exit 1
+    fi
+}
+
+
+if [[ -f "${SHARED_DIR}/kubeconfig" ]] ; then
+    export KUBECONFIG=${SHARED_DIR}/kubeconfig
+fi
+
+# Setup proxy if it's present in the shared dir
+if [[ -f "${SHARED_DIR}/proxy-conf.sh" ]]; then
+    # shellcheck disable=SC1091
+    source "${SHARED_DIR}/proxy-conf.sh"
+fi
+
+# Get the target upgrades release, by default, OPENSHIFT_UPGRADE_RELEASE_IMAGE_OVERRIDE is the target release
+# If it's serial upgrades then override-upgrade file will store the release and overrides OPENSHIFT_UPGRADE_RELEASE_IMAGE_OVERRIDE
+# upgrade-edge file expects a comma separated releases list like target_release1,target_release2,...
+export TARGET_RELEASES=("${OPENSHIFT_UPGRADE_RELEASE_IMAGE_OVERRIDE}")
+if [[ -f "${SHARED_DIR}/upgrade-edge" ]]; then
+    release_string="$(< "${SHARED_DIR}/upgrade-edge")"
+    # shellcheck disable=SC2207
+    TARGET_RELEASES=($(echo "$release_string" | tr ',' ' '))
+fi
+echo "Upgrade targets are ${TARGET_RELEASES[*]}"
+
+
+# Target version oc will be extract in the /tmp/client directory, use it first
+mkdir -p /tmp/client
+export OC_DIR="/tmp/client"
+export PATH=${OC_DIR}:$PATH
+
+
+export TARGET="${TARGET_RELEASES[0]}"
+TARGET_VERSION="$(env "NO_PROXY=*" "no_proxy=*" oc adm release info "${TARGET}" --output=json | jq -r '.metadata.version')"
+extract_oc
+
+check_target_version
```

Comment:
A minor suggestion, how about extracting the code in the function here to avoid introducing the function.

Reply:
if the step will not do other works in the future, I am ok to remove to function.